internal/uvm/mem_allocator.go

```diff
@@ -0,0 +1,182 @@
+package uvm
+
+import (
+	"github.com/pkg/errors"
+)
+
+const (
+	MemCls1MB uint32 = iota
+	MemCls4MB
+	MemCls16MB
+	MemCls64MB
+	MemCls256MB
+	MemCls1GB
+	MemCls4GB
+)
+
+const (
+	MegaByte  = 1024 * 1024
+	MemClsNum = MemCls4GB + 1
```

Comment:
I think this should be more explicit since it represents the memory class limit. 

I also think that we can go ahead and call this MemoryClass* (so like MemoryClass4GB, MemoryClassCount). We don't really gain much by using `memCls`.